### ISSUE TITLE
Wayland support woes (meta issue)

### ISSUE THREAD

**Issue by primeos** [telegramdesktop/tdesktop#7547]
tl;dr: I'm using `tdesktop` on Wayland (Sway) for approx. half a year now and it used to work pretty well. However, recently a few additional regressions have been sneaking in (see below). Therefore it might be a good idea to do some basic testing on Wayland as well (if possible).

### Current problems/regressions

- The first launch after an update results in a frozen `tdesktop` window that crashes after a few seconds
  - I think this issue occurs due to the changelog (or the notification for it) that is displayed only during the first launch. This only happens once but I'd still consider it an important issue.
- Quitting `tdesktop` results in a segmentation fault
  - This doesn't seem to have any consequences and only happens when terminating the application (-> minor issue), but it's still a bit unexpected.
- Right-clicking on a contact in the left sidebar results in the following error messages (but IIRC it still works as expected):
  ```
  qt.qpa.wayland: Non-toplevel surfaces can't request window states
  qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
  qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
  ```
- It seems like the current approach is more focused on ignoring Wayland for now:, e.g. https://github.com/telegramdesktop/tdesktop/commit/1a2b2c15c5ade505e94a005f1e002e8d951c380c (no offence intended, but I want to quickly mention this because it seems to be an important part of the problem)

### Why is Wayland support important?

- [Many Linux distributions already default to Wayland](https://en.wikipedia.org/wiki/Wayland_(display_server_protocol)#Adoption)
- [X11 development decreases and current efforts are focused on Wayland](https://www.phoronix.com/scan.php?page=news_item&px=X.Org-Server-2019-Low)

### Steps to reproduce

1. Use a Wayland compositor
2. Launch `telegram-desktop`
3. Check for a console message like `Using Wayland-EGL`

### Expected behaviour

Telegram-Desktop works on Wayland without crashing on the first launch after an update and without a segmentation fault after exiting the application.

### Actual behaviour

The Wayland support is currently degrading and far behind the X11 support.

### Configuration
**Operating system:** GNU/Linux (NixOS)

**Version of Telegram Desktop:** 2.0.1

**Used theme**: default

**Comment by AliDjango** [comment]
where did you download telegram from?binaries or snap package?

**Comment by primeos** [comment]
@AliDjango I built it from source / used the package of my Linux distribution: [tdesktop/default.nix](https://github.com/NixOS/nixpkgs/blob/606baf03ba204f7b205cb14c3e73d2f05aa22795/pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix)

Most Linux users will probably use a package provided by their Linux distribution (which unfortunately might vary quite a lot, though packaging `tdesktop` got much better recently).

**Comment by ilya-fedin** [comment]
The most of issues are due to poor support of Wayland in Qt :(

**Comment by ilya-fedin** [comment]
Please, create a separate issue for each problem that you have and attach a backtrace for each crash.
Errors in console are not a tdesktop issue, but warnings about unimplemented methods in Qt.
Thanks,
